linodecli/plugins/obj.py

```diff
@@ -63,9 +63,22 @@
 
 # Files larger than this need to be uploaded via a multipart upload
 UPLOAD_MAX_FILE_SIZE = 1024 * 1024 * 1024 * 5
-# This is how big the chunks of the file that we upload will be
+# This is how big (in MB) the chunks of the file that we upload will be
 # This is a float so that division works like we want later
-MULTIPART_UPLOAD_CHUNK_SIZE = 1024 * 1024 * 1024 * 5.0
+MULTIPART_UPLOAD_CHUNK_SIZE_DEFAULT = 1024.0
+
+def restricted_float_arg_type(max, min=0.0):
```

Comment:
There should probably be a non-zero default for min; I can pass bad values to `--chunk-size` to get interesting errors:

```bash
wsmith@oth-mpqrg::linode-cli[obj-file-upload-chunk-size]$ linode-cli obj put openapi.yaml wsmith-test --chunk-size 0
Traceback (most recent call last):
  File "/usr/local/bin/linode-cli", line 33, in <module>
    sys.exit(load_entry_point('linode-cli', 'console_scripts', 'linode-cli')())
  File "/Users/wsmith/source/linode-cli/linodecli/__init__.py", line 509, in main
    plugins.invoke(parsed.command, plugin_args, context)
  File "/Users/wsmith/source/linode-cli/linodecli/plugins/__init__.py", line 34, in invoke
    plugin.call(args, context)
  File "/Users/wsmith/source/linode-cli/linodecli/plugins/obj.py", line 852, in call
    COMMAND_MAP[parsed.command](get_client, args)
  File "/Users/wsmith/source/linode-cli/linodecli/plugins/obj.py", line 269, in upload_object
    _do_multipart_upload(bucket, filename, file_path, file_size, policy, chunk_size)
  File "/Users/wsmith/source/linode-cli/linodecli/plugins/obj.py", line 295, in _do_multipart_upload
    num_chunks = int(math.ceil(file_size / chunk_size))
ZeroDivisionError: float division by zero
```

and 

```bash
wsmith@oth-mpqrg::linode-cli[obj-file-upload-chunk-size]$ linode-cli obj put openapi.yaml wsmith-test --chunk-size 0.1
openapi.yaml (9 parts)
 Part 1
Upload failed!  Cleaning up!
Traceback (most recent call last):
  File "/Users/wsmith/source/linode-cli/linodecli/plugins/obj.py", line 304, in _do_multipart_upload
    upload.upload_part_from_file(
  File "/usr/local/lib/python3.9/site-packages/boto/s3/multipart.py", line 257, in upload_part_from_file
    key.set_contents_from_file(fp, headers=headers, replace=replace,
  File "/usr/local/lib/python3.9/site-packages/boto/s3/key.py", line 1285, in set_contents_from_file
    md5 = self.compute_md5(fp, size)
  File "/usr/local/lib/python3.9/site-packages/boto/s3/key.py", line 1036, in compute_md5
    hex_digest, b64_digest, data_size = compute_md5(fp, size=size)
  File "/usr/local/lib/python3.9/site-packages/boto/utils.py", line 1000, in compute_md5
    return compute_hash(fp, buf_size, size, hash_algorithm=md5)
  File "/usr/local/lib/python3.9/site-packages/boto/utils.py", line 1019, in compute_hash
    s = fp.read(size)
TypeError: argument should be integer or None, not 'float'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/linode-cli", line 33, in <module>
    sys.exit(load_entry_point('linode-cli', 'console_scripts', 'linode-cli')())
  File "/Users/wsmith/source/linode-cli/linodecli/__init__.py", line 509, in main
    plugins.invoke(parsed.command, plugin_args, context)
  File "/Users/wsmith/source/linode-cli/linodecli/plugins/__init__.py", line 34, in invoke
    plugin.call(args, context)
  File "/Users/wsmith/source/linode-cli/linodecli/plugins/obj.py", line 852, in call
    COMMAND_MAP[parsed.command](get_client, args)
  File "/Users/wsmith/source/linode-cli/linodecli/plugins/obj.py", line 269, in upload_object
    _do_multipart_upload(bucket, filename, file_path, file_size, policy, chunk_size)
  File "/Users/wsmith/source/linode-cli/linodecli/plugins/obj.py", line 310, in _do_multipart_upload
    raise upload_exception
TypeError: exceptions must derive from BaseException
```

Also, `float` accepts things that aren't _really_ numbers (per how floating point numbers work):

```bash
wsmith@oth-mpqrg::linode-cli[obj-file-upload-chunk-size]$ linode-cli obj put openapi.yaml wsmith-test --chunk-size NaN
Traceback (most recent call last):
  File "/usr/local/bin/linode-cli", line 33, in <module>
    sys.exit(load_entry_point('linode-cli', 'console_scripts', 'linode-cli')())
  File "/Users/wsmith/source/linode-cli/linodecli/__init__.py", line 509, in main
    plugins.invoke(parsed.command, plugin_args, context)
  File "/Users/wsmith/source/linode-cli/linodecli/plugins/__init__.py", line 34, in invoke
    plugin.call(args, context)
  File "/Users/wsmith/source/linode-cli/linodecli/plugins/obj.py", line 852, in call
    COMMAND_MAP[parsed.command](get_client, args)
  File "/Users/wsmith/source/linode-cli/linodecli/plugins/obj.py", line 269, in upload_object
    _do_multipart_upload(bucket, filename, file_path, file_size, policy, chunk_size)
  File "/Users/wsmith/source/linode-cli/linodecli/plugins/obj.py", line 295, in _do_multipart_upload
    num_chunks = int(math.ceil(file_size / chunk_size))
ValueError: cannot convert float NaN to integer
```

and

```bash
wsmith@oth-mpqrg::linode-cli[obj-file-upload-chunk-size]$ linode-cli obj put openapi.yaml wsmith-test --chunk-size infinity
usage: linode-cli obj put [-h] [--acl-public] [--chunk-size CHUNK_SIZE] FILE [FILE ...] BUCKET
linode-cli obj put: error: argument --chunk-size: Value must be between 0.0 and 5120.0
```

and 

```bash
wsmith@oth-mpqrg::linode-cli[obj-file-upload-chunk-size]$ linode-cli obj put openapi.yaml wsmith-test --chunk-size 1e4
usage: linode-cli obj put [-h] [--acl-public] [--chunk-size CHUNK_SIZE] FILE [FILE ...] BUCKET
linode-cli obj put: error: argument --chunk-size: Value must be between 0.0 and 5120.0
```

Why allow a fractional chunk size at all?  Might make more sense to either only accept whole numbers of MB or to accept units alongside the number (i.e. `--chunk-size 15KB`, or `--chunk-size 1B`).